core/templates/dev/head/domain/collection/CollectionObjectFactory.js

```diff
@@ -39,6 +44,12 @@ oppia.factory('CollectionObjectFactory', [
         var explorationId = this._nodes[i].getExplorationId();
         this._explorationIdToNodeIndexMap[explorationId] = i;
       }
+
+      // Populates skills.
+      for (var skillId in collectionBackendObject.skills) {
```

Comment:
I don't think this is quite right. Since we aren't using ECMA6, for-each loops may not do exactly what you expect. 

https://stackoverflow.com/questions/9329446/for-each-over-an-array-in-javascript

I suspect you may want to check the additional ``hasOwnProperty`` here to avoid looping over internal values of the skills object.